Vagrantfile

```diff
@@ -55,13 +55,13 @@ Vagrant.configure("2") do |config|
 
     # VirtualBox settings
     worker.vm.provider :virtualbox do |vb|
-      vb.customize ["modifyvm", :id, "--memory", "3048"]
+      vb.customize ["modifyvm", :id, "--memory", "5120"]
```

Comment:
Running the spark tests was causing the problem -- I still get OOM errors sometimes even with this so I think there might be some more investigation necessary